Formula/libarchive.rb

```diff
@@ -43,8 +42,11 @@ def install
     system "make", "install"
 
     # fixes https://github.com/libarchive/libarchive/issues/1819
-    inreplace lib/"pkgconfig/libarchive.pc", "Libs.private: ", "Libs.private: -liconv " if OS.mac?
-    inreplace lib/"pkgconfig/libarchive.pc", "Requires.private: iconv", ""
+    if OS.mac?
+      inreplace lib/"pkgconfig/libarchive.pc", "Libs.private: ", "Libs.private: -liconv "
+      inreplace lib/"pkgconfig/libarchive.pc", "Requires.private: iconv", ""
+    end
+
```

Comment:
Only one is supposed to be mac only

Reply:
confirmed the linux build patch is not needed, just removed it.

```
runner@fv-az593-769:/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ cat /home/linuxbrew/.linuxbrew/Cellar/libarchive/3.7.1/lib/pkgconfig/libarchive.pc
...
Libs.private: -lexpat -llzma -lzstd -llz4 -lb2 -lbz2 -lz
Requires.private:
```